_posts/2021-01-26-converting-puppetized-dsc-modules.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+title: "Converting to use Puppetized DSC Modules"
+author: pmcmaw
+categories:
+  - dsc
+  - puppet
+  - modules
+tags:
+  - tooling
+  - dsc resources
+  - windows
+  - modules
+  - forge
+  - powershell gallery
+---
+
+## Why bother?
+
+The main reason we are advocating moving to this new and shiny method is that it provides you with a long list of advantages and new features. 
+We're literally just taking the underyling DSC resources, wrapping them and making them available to you.
+If there’s a problem calling the DSC resources from Puppet, we’ll work on it, but if there’s a problem with how those resources behave, that’s upstream.
+
+**Seriously though what is there to not love?**
+
+* Minimal module maintainence costs
+* Automated module building
+* Property by property reporting
+* VSCode Extension with intellisense
+* We have 68 modules on the Forge in the [DSC namespace][dsc-namespace]
+* Converting is simple
+
+## So how can I do this?
+*Old method:* 
+```
+mod 'puppetlabs-dsc', '1.9.4'
+```
+
+*New method:*
+```
```

Comment:
I've just realized, we need to rebuild all of these modules once we release `Puppet.Dsc` with your side-by-side changes, so the numbers for these will all look a bit different.

Additionally, these versions _should_ look like `3.1.0-0-0` - the fourth digit needs to follow a dash and the fifth digit references our build increment, which for all of these will be `1` or higher after we rebuild 😬

Reply:
Once released, I plan on going through updating and validating against the Forge :-)